main/provider/index.js

```diff
@@ -321,21 +321,28 @@ class Provider extends EventEmitter {
     }, targetChain)
   }
 
-  sendTransaction (payload, res) {
-    const rawTx = this.getRawTx(payload)
-    const activeAccount = accounts.current()
+  fillTransaction (rawTx, cb) {
+    delete rawTx.gas
```

Comment:
I think `getRawTx()` already does this and also creates a copy of the tx rather than modifying the one passed in